Services/Storage.php

```diff
@@ -41,25 +12,63 @@ public function addElement(string $tableName, Model $model): void
      */
     public static function loadElements(string $tableName): array
     {
-        $filename = storage_path($tableName . ".json");
+        $filename = storage_path($tableName . '.' . CONFIG['currentStorageMethod']);
         if (!file_exists($filename)) {
             return [];
         }
 
         $data = file_get_contents($filename);
+        if(CONFIG['currentStorageMethod'] === 'xml'){
+            return (array)json_decode(json_encode(xmlrpc_decode($data)), true);
+        }
         return (array)json_decode($data, true);
     }
 
     /**
      * save elements to the table
      *
      * @param string $tableName
-     * @param string $jsonEncodedData
+     * @param string $encodedData
+     * @param string $extension
      * @return void
      */
-    public static function saveElements(string $tableName, string $jsonEncodedData): void
+    public static function saveElements(string $tableName, string $encodedData, string $extension = 'json'): void
     {
-        $filename = storage_path($tableName . ".json");
-        file_put_contents($filename, $jsonEncodedData);
+        $filename = storage_path($tableName . '.' . $extension);
+        file_put_contents($filename, $encodedData);
+    }
+
+    /**
+     * function for changing appropriate files with data to correct format as stated in config file
+     *
+     * @return void
+     */
+    public static function changeDataFilesToCorrectFormat(): void
+    {
+        $tableNames = ['Products', 'Employees', 'BranchOffice'];
+        $tableContents = [];
+
+        foreach ($tableNames as $tableName) {
+            foreach (CONFIG['possibleStorageMethods'] as $extension) {
+                $dir = storage_path($tableName . '.' . $extension);
+                if (file_exists($dir)) {
```

Comment:
You can combine those two `if` statements together

Reply:
combined